addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/ui/TabbedOutputPanel.java

```diff
@@ -148,35 +151,48 @@ private void addNewOutputSource(String name) {
                         ? registeredOutputSources.get(name).getAttributes()
                         : Map.of();
 
-        var outputPanel = new AbstractPanel();
-        outputPanel.setName(name);
-        outputPanel.setLayout(new BorderLayout());
+        var outputTab = new AbstractPanel();
+        outputTab.setName(name);
+        outputTab.setLayout(new BorderLayout());
         Icon icon =
                 attributes.containsKey(ATTRIBUTE_ICON)
                                 && attributes.get(ATTRIBUTE_ICON) instanceof Icon
                         ? (Icon) attributes.get(ATTRIBUTE_ICON)
                         : DOC_ICON;
-        outputPanel.setIcon(icon);
+        outputTab.setIcon(icon);
 
-        ChangeListener changeListener = e -> markTabRead(outputPanel, icon);
+        ChangeListener changeListener =
+                e -> {
+                    if (outputTab.isShowing()
+                            && outputTab.equals(tabbedPanel.getSelectedComponent())) {
+                        markTabRead(outputTab, icon);
+                    }
+                };
         tabbedPanel.addChangeListener(changeListener);
         outputSourceChangeListeners.put(name, changeListener);
 
-        ZapTextArea txtOutput = buildOutputTextArea(outputPanel, icon);
+        ZapTextArea txtOutput = buildOutputTextArea(outputTab, icon);
         JToolBar toolBar = buildToolbar(txtOutput, attributes);
-        outputPanel.add(toolBar, BorderLayout.PAGE_START);
+        toolBar.addMouseListener(
+                new java.awt.event.MouseAdapter() {
+                    @Override
+                    public void mouseEntered(MouseEvent e) {
+                        markTabRead(outputTab, icon);
+                    }
+                });
```

Comment:
Use a lambda?

Reply:
Can't, `MouseListener` isn't a functional interface (has more than one method to override / implement).

Reply:
Okay